compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -227,6 +229,14 @@ public TypeChecker(CompilerContext context) {
         this.typeParamAnalyzer = TypeParamAnalyzer.getInstance(context);
         this.anonymousModelHelper = BLangAnonymousModelHelper.getInstance(context);
         this.semanticAnalyzer = SemanticAnalyzer.getInstance(context);
+        initModifierFunctions();
+    }
+
+    private void initModifierFunctions() {
+        this.modifierFunctions.add("push");
+        this.modifierFunctions.add("pop");
+        this.modifierFunctions.add("shift");
```

Comment:
Have a small doubt regarding `shift()` on a tuple, created a spec issue - https://github.com/ballerina-platform/ballerina-spec/issues/474.

Reply:
I think that is just a extention of this. Spec says known to be fixed length we should error out. The example given we cannot properly define right ?

Reply:
I think we can improve this based on the spec discussion.

Can you also check the runtime behaviour?

Reply:
added case I :  by 714edb48df99b7d411f8100bd5df60c0b2e2b05b

Reply:
Improved for case I 714edb48df99b7d411f8100bd5df60c0b2e2b05b